worlds/apquest/test/test_hard_mode.py

```diff
@@ -0,0 +1,106 @@
+            self.assertAccessDependency(
+                ["Right Room Enemy Drop"],
+                [["Shield"], ["Health Upgrade"]],
+                only_check_listed=True,
+            )
+
+        # This tests that:
+        # 1. No Shield & No Health Upgrades   ->   Right Room Enemy Drop is not reachable.
+        # 2. Shield & No Health Upgrades      ->   Right Room Enemy Drop is reachable.
+        # 3. No Shield & All Health Upgrades  ->   Right Room Enemy Drop is reachable.
+
+        # Note: Every other item that isn't the Shield nor a Health Upgrade is collected into state.
+        # This even includes pre-placed items, which notably includes any event location/item pairs you created.
+        # In our case, it means we don't have to mention the Sword. By omitting it, it's assumed that we have it.
+
+        # This explains why the possible_items parameter is a list, but not why it's a list of lists.
+        # Let's look at the Final Boss Location. This location requires Sword, Shield, and both Health Upgrades.
+        # We could implement it like this:
+        with self.subTest("Test that the final boss isn't beatable without Sword, Shield, and both Health Upgrades"):
+            self.assertAccessDependency(
+                ["Final Boss Defeated"],
+                [["Sword", "Shield", "Health Upgrade"]],
+                only_check_listed=True,
+            )
+
+        # This would now test the following:
+        # 1. Without Sword, nor Shield, nor any Health Upgrades, the final boss is not beatable.
+        # 2. With Sword, Shield, and all Health Upgrades, the final boss is beatable.
+
+        # But, it's not really advisable to do this.
+        # Think about it: If we implemented our logic incorrectly and forgot to add the Shield requirement,
+        # this call would still pass. We'd rather make sure that each item individually is required:
+        for item in ["Sword", "Shield", "Health Upgrade"]:
+            with self.subTest(f"Test that the final boss requires {item}"):
+                self.assertAccessDependency(
+                    ["Final Boss Defeated"],
+                    [[item]],
+                    only_check_listed=True,
+                )
+
+        # This now tests that:
+        # 1. Without Sword, you can't beat the Final Boss
+        # 2. With Sword, you can beat the Final Boss (if you have everything else)
+        # 3. Without Shield, you can't beat the Final Boss
+        # 4. With Shield, you can beat the Final Boss (if you have everything else)
+        # 5. Without Health Upgrades, you can't beat the Final Boss
+        # 6. With all Health Upgrades, you can beat the Final Boss (if you have everything else)
+
+        # 2., 4., and 6. are the exact same check, so it is a bit redundant.
+        # But crucially, we are ensuring that all three items are actually required.
+
+        # This explains why the list can have multiple elements, but not why each element is also a list itself.
```

Comment:
So, I think of the list of lists as an OR list containing AND lists. Is that correct? Would that simplify any of this explanation?

(tbh the fact that it takes _this_ much effort to explain assertAccessDependency seems like clear evidence we need simpler, more intuitive testing APIs, and I'll probably keep using my own custom test helpers instead, but that's clearly out of scope here)

Reply:
assertAccessDependency is really clear when you use it for a single item :D I wonder if I should just remove all this and be like "Just accept that it's [["Item Name"]], unless you really need to know, in which case read this long and complicated explanation of what each list does elsewhere"

Reply:
And yes, correct. It's basically OR(AND()), but the AND() is a bit weird in that it allows you to do "reduction" to "one representative item"